project_milestone/views/project_task.xml

```diff
@@ -0,0 +1,75 @@
+<odoo>
+  <!-- List View -->
+  <record model="ir.ui.view" id="view_task_enhancement_milestone_tree" >
+    <field name="name">project.task.enhancement.milestone.tree</field>
+    <field name="model">project.task</field>
+    <field name="inherit_id" ref="project.view_task_tree2"/>
+    <field eval="2" name="priority"/>
+    <field name="arch" type="xml">
+      <field name="user_id" position="after">
+        <field name="milestone_id"
+               invisible="1"/>
+      </field>
+    </field>
+  </record>
+
+  <!-- Form View -->
+  <record model="ir.ui.view" id="view_task_enhancement_milestone_form" >
+    <field name="name">project.task.enhancement.milestone.form</field>
+    <field name="model">project.task</field>
+    <field name="inherit_id" ref="project.view_task_form2"/>
+    <field eval="2" name="priority"/>
+    <field name="arch" type="xml">
+      <data>
+        <xpath expr="//field[@name='user_id']" position="after">
+          <field name="use_milestones"
+                 invisible="1"/>
+          <field name="milestone_id"
+                 domain="[('project_id', '=', project_id)]"
+                 widget="selection"

```

Comment:
@patrickrwilson Remove `widget="selection"` and use `option={no_edit_create=True}`